src/config.rs

```diff
@@ -0,0 +1,207 @@
+use std::env;
+use std::fs;
+use std::io::{Read, Write};
+use std::path::PathBuf;
+
+use ansi_term::{Colour, Style};
+use toml;
+use xdg::BaseDirectories;
+
+use error::TealdeerError::{self, ConfigError};
+
+pub const SYNTAX_CONFIG_FILE_NAME: &'static str = "syntax.toml";
```

Comment:
Personal preference is to have special purpose config files, because it isolates handling code. With a shared config file there is always the consideration that unrelated logic is influenced. Of course there is a limit, 100 config files are not useful, so far it looks like there will be 2 or so config files with #44 also doing configuration.

Reply:
I'd go the other route, if that's OK for you. Start with a single config file, that's much easier to handle for the users, and also for the handling inside the code (you get a single configuration object, and have to deal with a single path, instead of having to handle multiple paths and have to do file-exists checks for every file etc).

If the config gets too big, it can still be split. But I don't see how the tealdeer config file should ever grow overly complex.

Inside the config, we could put all color configuration inside a `style` section for grouping. I guess it would look like this:

```toml
[style]

    [style.highlight]
    foreground = "blue"
    underline = false
    bold = true

    [style.description]
    underline = false
    bold = false

    [style.example_text]
    underline = false
    bold = false

    [style.example_code]
    underline = false
    bold = false

    [style.example_variable]
    underline = true
    bold = false
```

Other sections could then be added for non-style configuration (like #44).

Actually when I look at the example, I have to say that the nested syntax looks kind of ugly... I took a look at the [alacritty](https://github.com/jwilm/alacritty) config, which uses YAML and it looks much nicer, roughly like this:

```yaml
# Configuration for tealdeer

# Configure styling of the different parts of the output.
style:

  # Style used for highlighting the command name
  highlight:
    foreground: blue
    underline: false
    bold: true

  # Style used for the description text
  description:
    underline: false
    bold: false

  # Style used for the example text
  example_text:
    underline: false
    bold: false

  # Style used for the example code
  example_code:
    underline: false
    bold: false

  # Style used for variables inside the example code
  example_variable:
    underline: true
    bold: false
```

What do you think? Thanks to serde, switching to YAML would probably be easy.